main/src/com/google/refine/model/Cell.java

```diff
@@ -193,6 +194,29 @@ static public Cell deserialize(
 
     @Override
     public String toString() {
+        // TODO this is kept like this for now, but it should rather be a string which makes the difference
+        // between the cell and the value clear (i.e. not be the same string representation as the value)
+        // and include a representation of the recon field too.
+        // (toString() should not be exposed to the end user and is only meant to be used for logging/debugging
+        // purposes)
```

Comment:
If this is an implementation effort thing, perhaps one of the following would help (we already depend on both Apache lang3 and Guava):

- https://commons.apache.org/proper/commons-lang/apidocs/org/apache/commons/lang3/builder/ReflectionToStringBuilder.html
- https://commons.apache.org/proper/commons-lang/apidocs/org/apache/commons/lang3/builder/RecursiveToStringStyle.html
- https://commons.apache.org/proper/commons-lang/apidocs/org/apache/commons/lang3/builder/ToStringBuilder.html
- https://javadoc.io/static/com.google.guava/guava/31.0.1-jre/com/google/common/base/MoreObjects.html#toStringHelper(java.lang.Class)

Reply:
I didn't leave this out because I thought it would be complicated to re-implement (the methods added in this PR are all generated by Eclipse), but rather because I am worried that the return value of this `toString()` method can currently make it to the end user in one way or another (for instance, maybe in expression preview). Therefore, changing this method could have a noticeable impact for users, which would be unintended.

Independently, I'd rather avoid relying on helpers which rely on reflection, as it probably comes with a performance penalty, or at least with a complexity that can be avoided easily in such a case.

Does that make sense?

Reply:
Wouldn't this be showing up as "Cell@1234" if the default `Object.toString()` were being invoked?

There are implementations in that list which don't use reflection. Or you could roll you own by hand.

We can circle back and fix it another time if you'd rather not do it now.

Reply:
> Wouldn't this be showing up as "Cell@1234" if the default Object.toString() were being invoked?

No, at the moment, because we are already overriding `toString()` (before this PR), this would simply show up as the cell value - which is why I suspect there could be places where we do that without noticing.